### PR TITLE
Migrate `Cucumber.js` from 6.x to 7.x

### LINKED ISSUE #44: Migration to cucumberJs 7.X
Currently the library is functional with the version 6.X of [cucumberJS](https://github.com/cucumber/cucumber-js).

However there is some breaking change on the version 7.X.

Currently on the version 6, we are reusing a the JSON export that is naturally provided by cucumberjs in order to parse it and export it into the different channel.

However on cucumberJs@7 the JSON export is deprecated, it means that we need to rebuild a parsable JSON from the new format provided: [cucumber message](https://github.com/cucumber/common/tree/master/messages/javascript).

### PR DESCRIPTION
**Pull request by roottool**
Close #44

## Overview

This PR is a `Cucumber.js` migration from 6.x to 7.x.
I referred to https://github.com/cucumber/cucumber-pretty-formatter. Especially the below code.

https://github.com/cucumber/cucumber-pretty-formatter/blob/7ebffdfdb2998ef47ac0acf7adcbd6c20680dc9b/src/index.ts#L134-L143

## 🐛 Bug fix

https://github.com/restqa/cucumber-export/blob/e4ab095c11bbc3b459e1925caa6ff452f013f4ca/src/reports/stream.js#L15

An error had occurred by `Buffer.from(JSON.stringify(result))` when a reject the promise if the stream has an issue. But the below code didn't work.

https://github.com/restqa/cucumber-export/blob/e4ab095c11bbc3b459e1925caa6ff452f013f4ca/src/reports/stream.js#L11-L13

Therefore I added `try-catch`.

### Test log

```bash
❯ npm run test

> @restqa/cucumber-export@0.1.9 test
> jest

 FAIL  src/reports/stream.test.js
  ● #report - STREAM › Reject the promise if the stream has an issue

    expect(received).rejects.toThrow(expected)

    Expected message: "[STREAM REPORT][ERROR] - The _write() method is not implemented"
    Received message: "The _write() method is not implemented"

          25 |     })
          26 | 
        > 27 |     config.instance.write(createBuffer(result))
             |                     ^
          28 | 
          29 |     config.instance.end(err => {
          30 |       // if (err) errorHandler(err)

          at src/reports/stream.js:27:21
          at StreamReport (src/reports/stream.js:16:10)
          at Object.<anonymous> (src/reports/stream.test.js:56:19)

      54 |     }
      55 | 
    > 56 |     return expect(StreamReport(config, result)).rejects.toThrow(new Error('[STREAM REPORT][ERROR] - The _write() method is not implemented'))
         |                                                         ^
      57 |   })
      58 | })
      59 | 

      at Object.toThrow (node_modules/expect/build/index.js:274:20)
      at Object.<anonymous> (src/reports/stream.test.js:56:57)
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,17 +9,14 @@
     "test:coverage": "jest --coverage",
     "lint": "standard index.js src/",
     "lint:fix": "standard --fix index.js src/",
-    "example": "cucumber-js -f example/formatter.js:restqa.log example/features",
+    "example": "cucumber-js -f ./example/formatter.js:restqa.log example/features",
```
Comment (roottool) [inline package.json:12]: > This does mean that if you want to point to a local formatter implementation (i.e. not a Node module) then you should ensure it's a relative path starting with `./`.
> [Migration - Formmaters](https://github.com/cucumber/cucumber-js/blob/main/docs/migration.md#formatters)

A local formatter using `Cucumber.js` needs a relative path starting with `./` from v7.x.

**Review by olivierodo — APPROVED** [review]
Hello 👋,
Overall the PR is great.
I'm adding @tony-go, the technical lead as a second reviewer.

File: src/reports/stream.js [hunk 2]
```diff
@@ -1,5 +1,9 @@
 const stream = require('stream')
 
+function createStreamErr (err) {
```
Comment (tony-go) [inline src/reports/stream.js:3]: 😍

**Review by tony-go — APPROVED** [review]
Good job 👍🏼 ! LGTM

**Pull request merged**